### DISCUSSION TITLE
개발 중 떠오르는 (PR범위 한정)사소한 추가 기능 아이디어들 모아두고, 이후에 다같이 한번에 살펴보면 어떨까요?

### DISCUSSION THREAD

**Discussion by greyHairChooseLife** [wanted-pre-onboarding-side-project/habit-tracker#38] [Ideas]
## 경험 근거

이번에 PR 할 개발 범위에 아래 항목이 있습니다.

> 현재 표시된 주가 ‘오늘’이 포함된 경우, 다음 주로는 이동할 수 없다.

이걸 보고서 '이번 주를 바라보고 있다면 Next 버튼이 눌러지지 않게 만들어야겠다.'라고 생각했거든요? 그런데 고민 좀 해 보다가 개발하지 않기로 했습니다.

**왜냐면 괜히 좀 잘 해보겠다고 만들었다가 다른 멤버분들은 개발하지 않은 것이 튀어나와서 리뷰 하기만 어렵게 만드는 것이 아닐까 해서요.**

그렇다고 이런 작은 내용들을 하나하나 다 discussion이나 issue에 올리기도 번거롭고요.

## 제안

1. (곧 생길) 위키의 PR 컨벤션에 다음 항목을 추가합니다.
    - `issue: TODO`에 적힌 것, 적힌 것을 개발하기 위한 최소한의 추가 내용만 개발한다.
    - 이 밖에는 PR을 리뷰 할 때 고려하지 않는다.

2. `issue: TODO`를 생성 할 때, `discussion: Ideas`도 열고 PR 진행 도중에 (개발 범위에 한정)작은 추가 기능 아이디어들을 모아 둔다. 해당 PR이 merge되면 그때 이 discussion을 다 같이 협의 후 즉시 `issue: enhancement`를 발행 한다.

3. 위키의 [협업 프로세스 요약](https://github.com/wanted-pre-onboarding-side-project/habit-tracker/wiki/%ED%98%91%EC%97%85-%ED%94%84%EB%A1%9C%EC%84%B8%EC%8A%A4-%EC%9A%94%EC%95%BD-(-issue-or-discussion-%3F-))에 다음 항목을 추가합니다.
    - (신설)기타 규칙 : 2번 항목의 내용

4. 위키의 [issue 사용법](https://github.com/wanted-pre-onboarding-side-project/habit-tracker/wiki/issue-%EC%82%AC%EC%9A%A9%EB%B2%95)에 다음 항목을 추가합니다.
    - 기타 규칙 : 2번 항목의 내용
    - enhancement 라벨 설명 : 시간 제한 없이 자유롭게 PR(필요 approve: 2)

**Comment by jiyeon2** [comment]
좋습니다~ 
사소한 기능의 경우 각자 해석이 다를 수 있어서
다같이 살펴볼 시간이 있으면 좋겠습니다~

**Comment by chadseok** [comment]
네 혹시 이 질문은 잘 이해가 잘 안되서 그런데 구두로 설명 가능하실까요??

그리고 저는 이 부분을 구현하였는데

> 현재 표시된 주가 ‘오늘’이 포함된 경우, 다음 주로는 이동할 수 없다.

<br>

@greyHairChooseLife 님이 말씀하신 이 부분이 어떤 의도로 말씀하신지 정확하게 파악하지 못하였습니다

> 왜냐면 괜히 좀 잘 해보겠다고 만들었다가 다른 멤버분들은 개발하지 않은 것이 튀어나와서 리뷰 하기만 어렵게 만드는 것이 아닐까 해서요.

**Reply by greyHairChooseLife** [reply]
>@greyHairChooseLife
왜냐면 괜히 좀 잘 해보겠다고 만들었다가 다른 멤버분들은 개발하지 않은 것이 튀어나와서 리뷰 하기만 어렵게 만드는 것이 아닐까 해서요.

요것은 그니까 저희가 기획 회의에서 협의했던 기능들은 저희 셋이 best practice를 뽑자고 했잖아요? 그래서 코드 자체를 리뷰하는 것도 있지만 동시에,

>"우리 멤버들의 코드 중에 어떤 것이 베스트에 속할까?"

처럼 **비교**도 해야 합니다.

그런데 만약 `issue: TODO`에 기록 되지 않은 내용이라면(상관이 있는 거라고 해도) 누구는 개발을 하고 누구는 하지않을텐데요, 왜냐면 이것은 참신한 아이디어라던지.. 그러니까 UX에 대한 고민이 드러나는 것이니까요.

이렇게 되면 서로 비교하기 어렵고, 아무래도 리뷰 해야 할 코드가 의미 없이(리뷰와 비교를 동시에 하기 때문에) 길어지는 비용이 발생합니다. 개발을 하는 사람 입장에서도 그렇고요.
